### ISSUE TITLE
Random "missing required parameter client_id" errors

### ISSUE THREAD

**Issue by simonmaass** [wavezync/nuxt-vue3-google-signin#11]
I really dont know what is happening... randomly i get "missing required parameter client_id" in console... one day it works.. the next it doesnt... 

![image](https://github.com/wavezync/nuxt-vue3-google-signin/assets/32599710/9a62dcd8-249c-468d-9c01-7c48b3832a10)

Any one else experiencing this... using this with nuxt 3

**Comment by pokaiCheng** [comment]
I am also experiencing the same thing in local. Had to restart the build for it to work. Haven't deployed to uat/staging but if this happens it can be concerning...

**Comment by kasvith** [comment]
Can you provide an example? this seems to be an issue

**Comment by pokaiCheng** [comment]
I followed exact steps in the instructions, adding the module and adding googleSignIn.clientId

However, like simonmaaas said, sometimes it works, sometimes it doesn't. Mine seems to not work on first build, then had to stop everything and restart then it works fine. So very difficult to replicate the bug. 

Just a question, I looked at the code file plugin.ts, and it's getting the signin from: 
const { googleSignIn } = useRuntimeConfig().public

So shouldnt the googlesignin not be inside public? 
So would the nuxt.config.ts be:
modules: [
    'nuxt-vue3-google-signin'
  ],
public: {
  googleSignIn: {
    clientId: 'CLIENT ID OBTAINED FROM GOOGLE API CONSOLE',
  }
}

But like I mentioned, it is working but for me just on first build it doesnt.

**Comment by kasvith** [comment]
its picking up the config from nuxt config here and injecting it to runtime

https://github.com/wavezync/nuxt-vue3-google-signin/blob/8b6af9f251e648566a32b38dbe1f6aad24d87b84/src/module.ts#L43-L45

can you specify nuxt versions? so i can take a look

**Comment by kasvith** [comment]
@pokaiCheng @simonmaass 

can you share a example repo with nuxt3 using your exact configuration for the plugin?

**Comment by pokaiCheng** [comment]
@kasvith Vue 3.3.4, nuxt 3.6.5

Unfortunately I don't have an example repo. I'll see if I can build a simple repo with same configs and see if I can replicate the error.

**Comment by kasvith** [comment]
It would be nice to have a example repo with your exact configuration

**Comment by simonmaass** [comment]
i am using the same versions as @pokaiCheng and also my config is as described above...

**Comment by kasvith** [comment]
Can you please provide a repo with your exact configuration related to the project

That would help me to understand whats the problem

**Comment by pokaiCheng** [comment]
I've created a branch from your working example and changed the configs and package to match my versions. Very difficult to replicate the bug but sometimes the bug appears. What I did is switching between branches: one with the googleSignIn.clientId in the nuxt.config and one doesn't, then sometimes the bug will appear.
Here's my branch: https://github.com/pokaiCheng/nuxt-vue3-google-signin-example/tree/missing-required-parameter-client-id

![image](https://github.com/wavezync/nuxt-vue3-google-signin/assets/81480222/d5f52910-4368-48c4-aa5c-4ea093dc78b4)

**Comment by kasvith** [comment]
Thanks @pokaiCheng will take a look

**Comment by simonmaass** [comment]
we still have it and we really dont know what to do....

**Comment by kasvith** [comment]
Hi @simonmaass i didn't have time for looking at this, i will give an update

**Comment by zola33dsf** [comment]
Hi, do you have any updates please ?

**Comment by kasvith** [comment]
@zola33dsf unfortunately i couldn't reproduce this yet. would you share a repo which i can reproduce this?

**Comment by zola33dsf** [comment]
Yes, 
It is random so if it's not happening the first time you can try to refresh the page : 

https://stackblitz.com/edit/github-7tvhqt?file=nuxt.config.ts,app.vue

**Comment by kasvith** [comment]
Hi @zola33dsf I could reproduce this, seems something is wrong with the injection symbol.

**Comment by zola33dsf** [comment]
After upgrading to the new release of **Nuxt 3.8.1**, it seems that the problem has disappeared.

**Comment by kasvith** [comment]
Thanks for report @zola33dsf 

Hi @simonmaass @pokaiCheng can you also check and verify this works in nuxt3.8.1 ?

**Comment by kasvith** [comment]
I opened a bug report in nuxt https://github.com/nuxt/nuxt/issues/24155

**Comment by zola33dsf** [comment]
Hi, after upgrading to the latest release of **Nuxt 3.9.0** the issue came back..

**Comment by abdulla-iqbal** [comment]
> Just a question, I looked at the code file plugin.ts, and it's getting the signin from: const { googleSignIn } = useRuntimeConfig().public
> 
> ## So shouldnt the googlesignin not be inside public? So would the nuxt.config.ts be: modules: [ 'nuxt-vue3-google-signin' ], public: { googleSignIn: { clientId: 'CLIENT ID OBTAINED FROM GOOGLE API CONSOLE', } }

@pokaiCheng got a valid point see [plugin.ts](https://github.com/wavezync/nuxt-vue3-google-signin/blob/main/src/runtime/plugin.ts)

@kasvith here is complete procedure to reproduce it.

1.Take any nuxt app, upgrade it to latest for now it's `3.9.0`
2. configure `nuxt.config.ts` 
> Or one can  Change IAM_GOOGLE_CLIENT_ID to NUXT_PUBLIC_GOOGLE_CLIENT_ID as [mentioned here](https://github.com/wavezync/nuxt-vue3-google-signin/blob/main/playground/.env.example)
```
export default defineNuxtConfig({
   ...,
    modules: [
    'nuxt-vue3-google-signin',
  ],
  googleSignIn: {
    clientId: process.env.IAM_GOOGLE_CLIENT_ID,
  }
  });
```

3. now  remove `node_modules` and run `npm i` or `npm ci` and run `npm run dev`, after Nitro finishes build, visit server url e.g. `http://localhost:3000/`, in console you'll be able to see Error
![image](https://github.com/wavezync/nuxt-vue3-google-signin/assets/149565693/441c7d4e-08b9-42d2-b7e4-7945d41e6be3)

**Comment by kasvith** [comment]
Hi @abdulla-iqbal this seems an upstream issue with nuxt dev server, already created an issue there

**Comment by abdulla-iqbal** [comment]
Is there any way to solve it ?

**Comment by kasvith** [comment]
Unfortunately we have to wait till Nuxt team to resolve this.

**Comment by abdulla-iqbal** [comment]
got it.

**Comment by kasvith** [comment]
Hi @zola33dsf still waiting on nuxt team's update. it is a problem with upstream nuxt

**Comment by mpizman** [comment]
Hey, are there any updates? I have this issue using nuxt: 3.13.0 and nuxt-vue3-google-signin: 0.0.11
@kasvith

**Comment by davehague** [comment]
Hey all, thanks for the thread.  I just confirmed that downgrading from  `"nuxt": "^3.13.2"` (latest) to `"nuxt": "3.12.4"` fixed the issue.  That doesn't help fix it going forward, but at least it's a workaround for those having the problem.   I think [this issue](https://github.com/wavezync/nuxt-vue3-google-signin/issues/26) is related too, that was actually the error I was getting but found the answer to my problem here.

**Comment by MacMillan13** [comment]
@davehague The `"3.12.4"` version doesn't fix the issue for me.

**Comment by kasvith** [comment]
I have released a new version https://github.com/wavezync/nuxt-vue3-google-signin/releases/tag/v0.0.13 can you check again?
